src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -230,6 +235,16 @@ private void updateTransactionWithLinksAndCompanyName(Transaction transaction,
         transactionService.updateTransaction(transaction, loggingContext);
     }
 
+    public Optional<String> getEntityNumberFromOverseasEntitySubmission(String submissionId) {
```

Comment:
I would have implemented this method differently - the optionality of an entity number being returned would be based upon whether this was an UPDATE submission (so in this case present) or a REGISTRATION submission (not present). So line 242 would be changed to check for null (using `Objects.isNull()`) and return either an `Optional.of(entityNumber)` or `Optional.empty()`.

Line 244 would then change to throw `new SubmissionNotFoundException("Can not determine submission type")`, precluding the need to do this in the `getSubmissionType()` method above. The logic in that method would then also change slightly.

Reply:
Yeah, I did not want to throw from this "get" method but have now followed your suggestion.